### ISSUE TITLE
Incorrect beginBody output location

### ISSUE THREAD

**Issue by philipzaengle** [craftcms/cms#7779]
### Description

In some cases, Craft does not inject  `{% html at beginBody %}` in the correct location. Our situation arose due to using AlpineJs to add focus traps for additional accessibility. Using ES6 arrow functions within an `x-init` attribute caused all HTML registered to be output at `beginBody` to be output directly after the `=>` arrow function.

For anyone else experiencing this issue, the best solution is to manually register the location of your body begging, via the [beginBody](beginbody) method like so:

```twig
<body>
   {{ beginBody() }}
</body?
```

### Example:
```twig
<html>
<body
    x-init="$watch('modalOpen', (newValue) => {
      // do something
      })"
    >
  {% html at beginBody %}
    Hello World
  {%  endhtml %}
</body>
</html>
```

### Will return:

```html
<body
    x-init="$watch('modalOpen', (newValue) =>  Hello World
 {
      // do something
      })"
>
</body>
```

### Steps to reproduce

1. Add the code above to any CraftCMS template.
2. View the source

### Additional info

- Craft version: 3.6.11.2
- PHP version: 7.4
- Database driver & version: mysql

**Comment by khalwat** [comment]
The relevant RegEx is here: https://github.com/craftcms/cms/blob/develop/src/web/twig/nodevisitors/EventTagAdder.php#L88

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just fixed for the next release.

**Comment by khalwat** [comment]
While this fixes this specific case in terms of using an arrow function, ￼I think to truly fix this, it needs to ignore `>` inside of single or double quotes as a match for terminating the tag.

**Comment by brandonkelly** [comment]
@khalwat That regex only works if there is an even number of quotes throughout the remainder of the text node, which won’t work in a variety of cases (e.g. `<body foo="{{ something }}">`, where the notes will be `<body foo="`, the dynamic output tag, and `">` (plus whatever comes after `>` until the next Twig tag). So it’s a bit more complex than that. If you have any specific real world examples we should be worried about, feel free to post them and I can see what we can do to guard against them.

**Comment by khalwat** [comment]
My examples would be contrived... but if it's Alpine, you can have `>` inside of a number of tags as a comparison operator such as [x-if](https://github.com/alpinejs/alpine#x-if), and the same with plain old JavaScript `onclick=""` handlers. 

I admit it's pretty edge case though.

**Comment by philipzaengle** [comment]
additionally, and almost 100% in jest I could include a regex within my alpine init attribute. consider:

```
<body 
  x-data="" 
  x-init=" () => { data.match(/<(.*?)>/) ? alert('wat') }"
>
```

you're welcome

**Comment by brandonkelly** [comment]
Alright alright, just implemented a much more thorough fix. It’s a bit involved and there’s a _slight_ chance it could cause a BC break, if a template is doing something weird like `<body foo={{ '"' }}bar">`, so I committed it to the `3.7.` branch.

**Comment by khalwat** [comment]
![giphy](https://user-images.githubusercontent.com/7570798/114257587-fa610480-998e-11eb-84ce-5d3d4b9ad424.gif)

**Comment by brandonkelly** [comment]
We just released Craft 3.6.12 with the fix for `=>`.
